commit c2a0d91db36a4c75eb1f7be39a1a9c658ce96803
Author: Jason King <jason.king@joyent.com>
Date:   2019-06-26T20:09:50+00:00 (3 months ago)
    
    OS-7633 sysinfo should report the encryption status of a system
    Reviewed by: Josh Wilsdon <jwilsdon@joyent.com>
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Approved by: Dan McDonald <danmcd@joyent.com>

diff --git a/src/sysinfo b/src/sysinfo
index e8c76c6b..f57db1a8 100644
--- a/src/sysinfo
+++ b/src/sysinfo
@@ -291,6 +291,15 @@ function get_zpool()
         local size=$(( $used + $available ))
         Zpool_size=$(($size / 1024 / 1024 / 1024))
         Zpool_creation=$(zfs get -Hpo value creation ${Zpool})
+        Zpool_encrypted=$(zfs get -Hpo value encryption ${Zpool})
+
+        # The encryption property can be 'off', 'on', or a specific mechanism
+        # We treat anything but 'off' as enabled.
+        if [[ "$Zpool_encrypted" == "off" ]]; then
+            Zpool_encrypted="false"
+        else
+            Zpool_encrypted="true"
+        fi
 
         get_zpool_disks ${Zpool}
         get_zpool_profile ${Zpool}
@@ -894,6 +903,7 @@ END
 if [[ ${ZONENAME} == "global" ]]; then
     if [[ -n ${Zpool} ]]; then
         echo '  "Zpool": "'${Zpool}'",'
+        echo '  "Zpool Encrypted": '${Zpool_encrypted}','
         echo '  "Zpool Disks": "'${Zpool_disks}'",'
         echo '  "Zpool Profile": "'${Zpool_profile}'",'
         echo '  "Zpool Creation":' ${Zpool_creation}','
